### DISCUSSION TITLE
target87 Building Blocks

### DISCUSSION THREAD

**Discussion by verberg** [cssbattle/forum#199] [Q&A] [answered]
Why do I have only 83.1% match width this solution? I really understand now nothing:

<p><p><style>*{background:#F3AC3C;aspect-ratio:1;border-radius:10px}*>*{width:100;background:#1A4341;margin:65;box-shadow:170px 70px#1A4341}p{width:50;margin:65 100;box-shadow:20px 120px#1A4341,-20px 10px 0 10px#1A4341,40px 110px 0 10px#1A4341}p+p{box-shadow:none;background:#F3AC3C;width:70;margin:-65 10
<img width="851" alt="Bildschirmfoto 2021-12-14 um 08 59 50" src="https://user-images.githubusercontent.com/94834403/145957047-869c7311-1427-4f40-b6ce-666e01f709ba.png">

**Comment by verberg** [comment, marked as answer]
I've seen the problem myself: * and border-radius:10px --> also the body has a border-radius. this was wrong.
My new solution with 100% mach:
<p><p><p><style>*{background:#F3AC3C}p{width:100;aspect-ratio:1;background:#1A4341;margin:65 57;box-shadow:180q 74q#1A4341;border-radius:10q}p+p{width:50;margin:-165 157;box-shadow:-42q 10q 0 10px#1A4341,21q 127q#1A4341,63q 116q 0 10px#1A4341}p+p+p{box-shadow:none;background:#F3AC3C;width:70;margin:165 157

**Reply by Beowolve** [reply]
You can reduce the size of this code by reusing your colors a lot. `box-shadow` for example does use the active text color by default, so you can set `color:1A4341` and the remove all color codes from `box-shadow`. You can even use box-shadow to paint the background color with the `inset` keyword. However it its possible to get 99.9% problems, you have been warned.

**Reply by verberg** [reply]
Thanks a lot for this idea!

**Reply by verberg** [reply]
Yes, you'r right. sometimes you have 100% match width a background-color, and if you want replace the background width box-shadow, you have only 99.9%. It's really cracy!
